src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -278,6 +322,83 @@ internal sealed override ImmutableArray<string> NotNullWhenFalseMembers
                 syntax.Body, syntax.ExpressionBody, syntax, diagnostics);
         }
 
+        /// <summary>
+        /// Constructor only for auto-properties
+        /// </summary>
+        private SourcePropertyAccessorSymbol(
```

Comment:
>SourcePropertyAccessorSymbol [](start = 16, length = 28)

There is a lot of code duplication here and it is going to be very difficult to see what is the difference. Please refactor to share as much as possible between all three constructors.